src/definitions/list.ts

```diff
@@ -0,0 +1,60 @@
+import { GetGen } from '../typegenTypeHelpers'
+import { AllNexusNamedTypeDefs } from './wrapping'
+import { NexusTypes, withNexusSymbol } from './_types'
+
+type AllNamedTypes = GetGen<'allNamedTypes', string> | AllNexusNamedTypeDefs
+
+/**
+ * list()
+ */
+
+type ListableTypes = AllNamedTypes | NexusNonNullDef<NonNullableTypes> | NexusListDef<ListableTypes>
+
+export type NexusListDefConfig<TypeName extends ListableTypes> = {
+  type: TypeName
+}
+
+export class NexusListDef<TypeName extends ListableTypes> {
+  // @ts-ignore
+  private _isNexusListDef: boolean = true
```

Comment:
Needed to differentiate it from the `NexusNonNullDef`, otherwise, as both classes have the same constructor, TS doesn't differentiate them